### ISSUE TITLE
Sending Notifications in bulk

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ntrees/javapns#22]
```
I'm trying to send push notifications in bulk, around 800 at a time. But none 
of the push notifications are being sent.I can successfully send push for 
around 100 devices at a time but not able to send to devices more than 200 at 
once.
I am maintaining a persistent connection with APNS.

Did anyone face this issue before? what is the solution to this problem?


```

Original issue reported on code.google.com by `sudheer....@gmail.com` on 8 Jul 2010 at 4:14

**Comment by GoogleCodeExporter** [comment]
```
  Hi. 
  Maybe apple has some limitation in the size of each request size. Other that send 800 device at once , you can use 4 thread . Each sends 200 instead.
 Hopefully It can give you some help.

```

Original comment by `feiyangf...@gmail.com` on 22 Jul 2010 at 9:58

**Comment by GoogleCodeExporter** [comment]
```
Hi Sudheer...can you share the code template to send notification in bulk. 
Please see the thread I started
http://code.google.com/p/javapns/issues/detail?id=35&can=1
```

Original comment by `pilotj...@gmail.com` on 9 Dec 2010 at 3:51

**Comment by GoogleCodeExporter** [comment]
```
Solved in 2.0.  The new javapns.notification.transmission.NotificationThread 
class provides the ability to limit and manage automatically the number of 
notifications per connection, to avoid this issue with Apple servers.  Further 
more, the new javapns.notification.transmission.NotificationThreads class 
provides the ability to spread the work over multiple NotificationThread 
threads.
```

Original comment by `sype...@gmail.com` on 9 Sep 2011 at 7:04
- Changed state: **Fixed**
